### PR TITLE
Add some text near the top of master_list_file.rst and generate a FATES version of the file

### PR DESCRIPTION
**Pull request by slevisconsulting**
### Description of changes
Adding code that writes a disclaimer and a few of the case's flags near the top of master_list_file.rst.
Adding code to generate a FATES version of master_list_file.rst.
Naming new file `master_list_file_fates.rst` and renaming existing file `master_list_file_nofates.rst`.

### Specific notes

Contributors other than yourself, if any:
@billsacks @ekluzek @dlawrenncar 

CTSM Issues Fixed (include github issue #):
Continues where #1209 and #1222 left off.

Are answers expected to change (and if so in what way)?
No, except in relation to master_list_file.rst.

Any User Interface Changes (namelist or namelist defaults changes)?
As explained above.

Testing performed, if any:
Ran `ERP_P36x2_D_Ld3.f10_f10_musgs.I1850Clm50BgcCrop.cheyenne_gnu.clm-extra_outputs` to confirm new disclaimer text appears at the top of master_list_file.rst.

### CODE REVIEW

File: doc/source/users_guide/setting-up-and-running-a-case/master_list_file.rst [hunk 1]
```diff
@@ -2,6 +2,8 @@
 CTSM History Fields
 ===================
 
+CAUTION: Not all variables are relevant / present for all CTSM cases.
+
```
Comment (slevisconsulting) [inline doc/source/users_guide/setting-up-and-running-a-case/master_list_file.rst:6]: Diffs in this file other this disclaimer are caused by changes that have occurred prior to this PR.

**Comment by slevisconsulting** [comment]
This PR is ready for review.
Also I want to start the test-suites but wanted to check whether I should generate new baselines, yet.
Thanks!

**Comment by slevisconsulting** [comment]
The documentation built successfully. I looked at it and it's mostly ok, so I pushed it. The only detail that I haven't been able to resolve is that the table of contents shows the old naming of the tables "CTSM History Fields" here:
`https://escomp.github.io/ctsm-docs/versions/master/html/users_guide/index.html`
even though it shows the new naming "CTSM History Fields (nofates)" and "CTSM History Fields (fates)" here:
`https://escomp.github.io/ctsm-docs/versions/master/html/users_guide/setting-up-and-running-a-case/index.html`

**Review by billsacks — APPROVED** [review]
This looks great to me – thanks a lot, @slevisconsulting !

**Comment by billsacks** [comment]
> The only detail that I haven't been able to resolve is that the table of contents shows the old naming of the tables "CTSM History Fields" here

I have run into things like that before when I do an incremental rebuild of the documentation instead of a clean build: sometimes links don't get updated properly when doing an incremental rebuild, presumably because the dependencies aren't smart enough. If you're feeling patient, the simple thing to do is to run `build_docs` with the `-c` argument (`--clean`). If you're feeling less patient, you can find the specific problem files in the doc build area and remove them before running `build_docs`.

**Comment by billsacks** [comment]
My tentative thought is that, given how limited the source code changes are here, this PR could be combined with #1482 for a bit-for-bit tag (but we could also keep them as two separate little tags). We'll talk about this in tomorrow morning's software meeting, then @slevisconsulting we can let you know the result of that planning.

**Comment by slevisconsulting** [comment]
> > The only detail that I haven't been able to resolve is that the table of contents shows the old naming of the tables "CTSM History Fields" here
> 
> I have run into things like that before when I do an incremental rebuild of the documentation instead of a clean build: sometimes links don't get updated properly when doing an incremental rebuild, presumably because the dependencies aren't smart enough. If you're feeling patient, the simple thing to do is to run `build_docs` with the `-c` argument (`--clean`). If you're feeling less patient, you can find the specific problem files in the doc build area and remove them before running `build_docs`.

Ok, done.

**Comment by slevisconsulting** [comment]
**Izumi test-suite:** OK, I think. Four tests returned NLCOMP failures like this:
```
Comparison failed between '/scratch/cluster/slevis/tests_0908-183947iz/ERI_D_Ld9_P48x1_Vnuopc.f10_f10_mg37.I2000Clm50Sp.izumi_nag.clm-SNICARFRC.GC.0908-183947iz_nag/CaseDocs/lnd_in' with '/fs/cgd/csm/ccsm_baselines/ctsm5.1.dev054/ERI_D_Ld9_P48x1_Vnuopc.f10_f10_mg37.I2000Clm50Sp.izumi_nag.clm-SNICARFRC/CaseDocs/lnd_in'
  missing variable: 'fatmlndfrc'
```

**Cheyenne test-suite:** IN PROGRESS

**Comment by billsacks** [comment]
> Four tests returned NLCOMP failures

In a situation like this where you clearly haven't made any changes that would impact these namelists, I would probably just ignore these, assuming there is something wrong with the baselines or the NLCOMP comparison code itself. Though you may want to double check that all of your externals were up to date when you ran the testing.

**Comment by slevisconsulting** [comment]
> Though you may want to double check that all of your externals were up to date when you ran the testing.

I reran /manage_externals/checkout_externals on izumi and resubmitted the test-suite but got the same four NLCOMP failures.

Cheyenne test-suite also OK.
(Here I got an expected NLCOMP failure due to the Fates test that I modified to output the master_list_fates.rst file.)

**Comment by billsacks** [comment]
Thanks, @slevisconsulting ! I will merge this shortly.

**Pull request merged**